fastlane/lanes/build.rb

```diff
@@ -15,17 +15,30 @@
   #####################################################################################
   desc 'Builds and updates for distribution'
   lane :build_and_upload_release do |options|
-    android_build_prechecks(skip_confirm: options[:skip_confirm], final: true) unless options[:skip_prechecks]
-    android_build_preflight() unless options[:skip_prechecks]
+    unless options[:skip_prechecks]
+      ensure_git_branch(branch: '^release/') unless is_ci
```

Comment:
This is not related to this PR since this is just moved from elsewhere, but I do wonder if we should really hardcode `is_ci` checks like this - especially since we already have a way to skip these checks in CI by passing `skip_prechecks`. My worry is that hardcoding checks like this makes it harder to reason about the callers without looking at the source code.

@AliSoftware we recently had a discussion about the "is CI" checks in Tumblr and I'd love to hear what you think about this in general.

P.S: I think this is a good place to bring up this discussion, but let's not make any changes to this PR related to this. If we decide removing the checks, we can do that in its own PR.

Reply:
Actually a wider question came up as part of @spencertransier 's work on the versioning model.

In particular, the various `*_prechecks` actions that we have in the `release-toolkit` are (1) kinda obscure in what they're doing/checking (2) too tied with assumptions about the version model and format used by the projects (3) should probably be reworked or broken into more atomic and specific actions (similar to calling the more atomic `ensure_git_branch` here instead of calling some obscure `android_*_prechecks` action).

So we might want to have a broader discussions about those prechecks in general.

To answer your question directly though, I agree that it makes more sense to just have a `options[:skip_prechecks]` test wrapping precheck code, and not using `unless is_ci`. We can then always call the lane with `skip_prechecks:true` within the `.buildkite/commands/*.sh` invoking that lane from CI steps if we want, which would make things more explicit and avoid having redundant flags.

Reply:
> (3) should probably be reworked or broken into more atomic and specific actions

That sounds like a great plan to me!

Reply:
>  do wonder if we should really hardcode `is_ci` checks like this - especially since we already have a way to skip these checks in CI by passing `skip_prechecks`

I had the same thought while breaking out these `*_prechecks` actions into the app Fastfiles. These hardcoded `is_ci` checks existed in the `*_prechecks` actions, but I didn't have any context for why they were there in the first place or why they are added to the same check in some of the precheck actions but not others. So overall, I agree that removing the hardcoded `is_ci`'s and moving them them to the `skip_prechecks` section is a great idea.

>  (3) should probably be reworked or broken into more atomic and specific actions

This is something I'm planning to work on after the env var project is finished 👍